File: backend/app/schemas/alert.py
"""
Alert schemas.
"""

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import AlertSeverity, AlertStatus


class AlertOut(BaseModel):
    id: uuid.UUID
    patient_id: uuid.UUID
    related_vital_id: uuid.UUID | None
    related_prediction_id: uuid.UUID | None
    title: str
    message: str
    severity: AlertSeverity
    status: AlertStatus
    acknowledged_at: datetime | None
    resolved_at: datetime | None
    emergency_contact_notified: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/alert.ts
export type AlertSeverity = "info" | "warning" | "critical";
export type AlertStatus = "new" | "acknowledged" | "resolved";

export interface Alert {
  id: string;
  patient_id: string;
  related_vital_id: string | null;
  related_prediction_id: string | null;
  title: string;
  message: string;
  severity: AlertSeverity;
  status: AlertStatus;
  acknowledged_at: string | null;
  resolved_at: string | null;
  emergency_contact_notified: boolean;
  created_at: string;
}
